### PR TITLE
Add support for JSON config files to built-in server runner

### PR DESCRIPTION
**Pull request by djmattyg007**

### CODE REVIEW

**Comment by mar10** [comment]
Could you give a use case and example? 
We also should add some documentation for this.

**Comment by djmattyg007** [comment]
Some people aren't going to need the full flexibility of a python-based configuration file. In fact, some people who want to use this aren't going to want to know how to write python at all. They just want an awesome webdav server that works and is easy to configure. JSON configuration is much more common and probably more well-understood, and given how closely it maps to python's lists and dictionaries, it seems like a great fit (especially given the amount of effort involved in supporting it is incredibly small).

I'll update this PR as soon as possible with some documentation, and an example JSON config file.

**Comment by mar10** [comment]
I would like to merge this. Do you think you'll find some time to add the sample docs?

File: wsgidav/server/run_server.py [hunk 1]
```diff
@@ -196,6 +196,11 @@ def _readConfigFile(config_file, verbose):
     if not os.path.exists(config_file):
         raise RuntimeError("Couldn't open configuration file '%s'." % config_file)
 
+    if config_file.endswith(".json"):
+        import json
```
Comment (mar10) [inline wsgidav/server/run_server.py]: I think we can move this import to the header

**Comment by djmattyg007** [comment]
I'm so sorry for forgetting about this. I'm working on the patch now!

**Comment by djmattyg007** [comment]
I've updated the patch to include a sample config file, and to mention this in the docs.

**Comment by mar10** [comment]
Thanks a lot!

**Pull request merged**